drivers/can/can_mcan.c

```diff
@@ -831,106 +833,104 @@ static int can_mcan_get_free_ext(volatile struct can_mcan_ext_filter *filters)
 	return -ENOSPC;
 }
 
-static int can_mcan_attach_ext(struct can_mcan_data *data,
-			       struct can_mcan_msg_sram *msg_ram,
-			       can_rx_callback_t isr, void *cb_arg,
-			       const struct zcan_filter *filter)
+static int can_mcan_add_rx_filter_ext(struct can_mcan_data *data,
+				      struct can_mcan_msg_sram *msg_ram,
+				      can_rx_callback_t callback, void *user_data,
+				      const struct zcan_filter *filter)
 {
 	struct can_mcan_ext_filter filter_element = {
 		.id2 = filter->id_mask,
 		.id1 = filter->id,
 		.eft = CAN_MCAN_EFT_MASKED
 	};
-	int filter_nr;
+	int filter_id;
 
 	k_mutex_lock(&data->inst_mutex, K_FOREVER);
-	filter_nr = can_mcan_get_free_ext(msg_ram->ext_filt);
+	filter_id = can_mcan_get_free_ext(msg_ram->ext_filt);
 
-	if (filter_nr == -ENOSPC) {
-		LOG_INF("No free extender id filter left");
+	if (filter_id == -ENOSPC) {
+		LOG_INF("No free extended id filter left");
 		return -ENOSPC;
 	}
 
 	/* TODO propper fifo balancing */
-	filter_element.efce = filter_nr & 0x01 ? CAN_MCAN_FCE_FIFO1 :
+	filter_element.efce = filter_id & 0x01 ? CAN_MCAN_FCE_FIFO1 :
 						 CAN_MCAN_FCE_FIFO0;
 
-	memcpy32_volatile(&msg_ram->ext_filt[filter_nr], &filter_element,
+	memcpy32_volatile(&msg_ram->ext_filt[filter_id], &filter_element,
 			  sizeof(struct can_mcan_ext_filter));
 	CACHE_CLEAN(&msg_ram->ext_filt[filter_nr],
 		    sizeof(struct can_mcan_ext_filter));
 
 	k_mutex_unlock(&data->inst_mutex);
 
-	LOG_DBG("Attached ext filter at %d", filter_nr);
+	LOG_DBG("Attached ext filter at %d", filter_id);
 
 	if (filter->rtr) {
-		data->ext_filt_rtr |= (1U << filter_nr);
+		data->ext_filt_rtr |= (1U << filter_id);
 	} else {
-		data->ext_filt_rtr &= ~(1U << filter_nr);
+		data->ext_filt_rtr &= ~(1U << filter_id);
 	}
 
 	if (filter->rtr_mask) {
-		data->ext_filt_rtr_mask |= (1U << filter_nr);
+		data->ext_filt_rtr_mask |= (1U << filter_id);
 	} else {
-		data->ext_filt_rtr_mask &= ~(1U << filter_nr);
+		data->ext_filt_rtr_mask &= ~(1U << filter_id);
 	}
 
-	data->rx_cb_ext[filter_nr] = isr;
-	data->cb_arg_ext[filter_nr] = cb_arg;
+	data->rx_cb_ext[filter_id] = callback;
+	data->cb_arg_ext[filter_id] = user_data;
 
-	return filter_nr;
+	return filter_id;
 }
 
-int can_mcan_attach_isr(struct can_mcan_data *data,
-			struct can_mcan_msg_sram *msg_ram,
-			can_rx_callback_t isr, void *cb_arg,
-			const struct zcan_filter *filter)
+int can_mcan_add_rx_filter(struct can_mcan_data *data,
+			   struct can_mcan_msg_sram *msg_ram,
+			   can_rx_callback_t callback, void *user_data,
+			   const struct zcan_filter *filter)
 {
-	int filter_nr;
+	int filter_id;
 
-	if (!isr) {
+	if (!callback) {
```

Comment:
```suggestion
	if (callback == NULL) {
```
To be consistent witch the change above

Reply:
Fixed.